drafts/builder-pattern-tables.md

```diff
@@ -0,0 +1,354 @@
+
+    <!-- Created Column -->
+    <ng-container matColumnDef="created-date">
+        <th mat-header-cell *matHeaderCellDef mat-sort-header disableClear>
+            Created
+        </th>
+        <td mat-cell *matCellDef="let row">{{row.created_at | date}}</td>
+    </ng-container>
+
+    <tr mat-header-row *matHeaderRowDef="displayedColumns"></tr>
+    <tr mat-row *matRowDef="let row; columns: displayedColumns;"></tr>
+</table>
+```
+
+### Shared Table
+
+As a first step, you might think about creating a shared component for your tables. This ensures that all tables within your application are consistent and easy to maintain. However, crafting such a shared table component is not as easy as it seems. The table would need to handle various data entities, cell formats, cell types, and date formats. And the more features you need to pack into your table, the more complex such a shared table can get. 
+
+Our shared table requires two inputs. First, the table configuration, consisting of headers, the data type for each column, information on how to retrieve the cell value from our data entity, and potential information about stylings and formatting of the cell value and columns.
+
+Second, the list of data entities or rather the row objects that we want to display in our table. Thereby, the table needs to be able to deal with various data types (when using Typescript at least, but why wouldn't you use Typescript in the first place).
+
+You can find an example of our shared table template at the bottom of this article.
+
+### Column Configuration
+Our first shared table input is the table configuration, or rather list of column configurations. Thereby, the column definition interface consists of four basic parts: the column title, a unique column ID (used for testing with cypress), the cell content configuration, and styling information.
+
+```typescript
+export type TableCellContentTypes =
+  | TableCellContentValue
+  | TableCellContentValueList
+  | TableCellContentIcon
+  | TableCellContentButton
+  | undefined;
+
+export type TableColumnConfiguration<T> = {
+  columnId: string;
+  title: string;
+  cellContent: {
+    displayType: TableCellDisplayingType;
+    valueGetterFn: (data: T) => TableCellContentTypes;
+    formatFn: (cellValue: TableCellContentTypes) => TableCellContentTypes;
+  };
+  styling: {
+    width?: string;
+    minWidth?: string;
+    maxWidth?: string;
+  };
+};
+```
+
+The `cellContent` attribute defines a value getter function specifying how to retrieve the required cell value from our data object. We can also add formatting functions here to e.g. transform the fetched value to a date or currency.
+
+The `displayType` attribute is used to switch between icons, buttons, string lists (multi-line), or simple string values displaying inside the table cell. Of course, this could easily be extended with other types that you want to display in your table cells, like references or even nested tables (please don't do that). 
```

Comment:
```suggestion
The `displayType` attribute is used to switch between icons, buttons, string lists (multi-line), or simple string values displaying inside the table cell. Of course, this could easily be extended with other types that you want to display in your table cells, like references or even nested tables (please don't do that).
```